spec/controllers/automated_tests_controller_spec.rb

```diff
@@ -8,14 +8,43 @@
       # TODO: write tests
     end
     context 'GET manage' do
-      before :each do
-        get_as user, :manage, params: params
-      end
-      it 'User should be able to view the Automated Testing manage page' do
-        expect(response.status).to eq(200)
+      context 'Manage page' do
+        before :each do
+          get_as user, :manage, params: params
+        end
+        it 'User should be able to view the Automated Testing manage page' do
+          expect(response.status).to eq(200)
+        end
+        it 'should render the assignment_content layout' do
+          expect(response).to render_template('layouts/assignment_content')
+        end
       end
-      it 'should render the assignment_content layout' do
-        expect(response).to render_template('layouts/assignment_content')
+      context 'Getting the test student grouping' do
```

Comment:
Your tests need to cover the following cases:

- the test student exists
    - the grouping exists
        - the membership exists
        - the membership does not exist
    - the grouping does not exist
        - the group exists
        - the group does not exist
- the test student does not exist

In every case where a new object needs to be created, you also need a test for what happens if the creation fails because of a validation error.